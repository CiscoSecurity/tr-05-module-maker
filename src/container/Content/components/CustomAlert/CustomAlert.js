import React from "react";
import "./CustomAlert.scss"
import * as Constants from "globals/constants/constants";
import Icons from "globals/icons/sprite.svg";

export const CustomAlert = ({title, message, closeAlertHandler}) => (
    <div className="alert">
        <div className="alertContent">
            <div className="alertHeader">
                {title}
                <svg className="closeIcon" onClick={closeAlertHandler}>
                    <use xlinkHref={`${Icons}#icon-small-x-close`}/>
                </svg>
            </div>
            <div className="alertBody">
                {message}
            </div>
            <div className="alertFooter">
                <button type="button"
                        className="closeButton"
                        onClick={closeAlertHandler}>
                    {Constants.CLOSE_BTN_LABEL}
                </button>
            </div>
        </div>
    </div>
)
