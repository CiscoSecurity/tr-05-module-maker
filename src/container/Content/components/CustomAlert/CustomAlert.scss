@import "src/globals/constants/colors";

.alert {
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;

  .alertContent {
    background-color: $white-color;
    margin: auto;
    border-radius: 6px;
    border: 1px solid $border-gray-color;
    min-width: 250px;
    width: 20%;
    color: $black-color;
    padding: 0;

    .alertHeader {
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 20px 20px 15px;
      box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
      background-color: #fff;
      border-top-left-radius: 6px;
      border-top-right-radius: 6px;
      font-size: 18px;
      font-weight: 300;
      line-height: 1.1;
      color: $header-text-color;

      .closeIcon {
        fill: $gray-color;
        width: 15px;
        height: 15px;
        margin: 4px;
        float: right;
        cursor: pointer;

        &:hover {
          fill: $hover-color-dark;
        }
      }
    }

    .alertBody {
      position: relative;
      text-align: left;
      flex: 1 1 auto;
      padding: 20px 20px 20px;
      line-height: 1.5;
      color: $header-text-color;

      &::first-letter {
        text-transform: capitalize;
      }
    }

    .alertFooter {
      display: flex;
      align-items: center;
      justify-content: flex-end;
      padding: 20px;
      background-color: $footer-color;
      border-bottom-left-radius: 6px;
      border-bottom-right-radius: 6px;

      .closeButton {
        color: $white-color;
        background-color: $blue-background;
        display: inline-block;
        text-align: center;
        vertical-align: middle;
        user-select: none;
        border: 1px solid transparent;
        border-radius: 3px;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        font-weight: 400;
        line-height: 1.5;
        height: 28px;
        padding: 0 15px;
        margin: 0;
        min-width: 75px;
        cursor: pointer;

        &:hover {
          background-color: $hover-color-dark;
        }
      }
    }
  }
}
