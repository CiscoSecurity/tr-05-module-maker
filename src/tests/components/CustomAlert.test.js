import { mount } from 'enzyme';
import { CustomAlert } from "../../container/Content/components/CustomAlert/CustomAlert";
import React from "react";
import toJson from "enzyme-to-json";


describe('CustomAlert', () => {
    it('should render CustomAlert', () => {
        const props = {title: 'Test title', message: 'Test message'}
        const wrapper = mount(<CustomAlert {...props}/>);

        expect(wrapper.find('div.alertHeader').text()).toBe('Test title');
        expect(wrapper.find('div.alertBody').text()).toBe('Test message');
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
