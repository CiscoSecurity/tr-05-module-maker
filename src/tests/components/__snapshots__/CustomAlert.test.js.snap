// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert should render CustomAlert 1`] = `
<CustomAlert
  message="Test message"
  title="Test title"
>
  <div
    className="alert"
  >
    <div
      className="alertContent"
    >
      <div
        className="alertHeader"
      >
        Test title
        <svg
          className="closeIcon"
        >
          <use
            xlinkHref="sprite.svg#icon-small-x-close"
          />
        </svg>
      </div>
      <div
        className="alertBody"
      >
        Test message
      </div>
      <div
        className="alertFooter"
      >
        <button
          className="closeButton"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</CustomAlert>
`;
